workbench-core/authorization/src/dynamicAuthorization/dynamoDBDynamicPermissionsPlugin.ts

```diff
@@ -0,0 +1,504 @@
+    }
+
+    const response = await query.execute();
+    const identityPermissions: IdentityPermission[] =
+      response.Items?.map((item) => {
+        return this._processItemToIdentityPermission(item);
+      }) ?? [];
+    return { identityPermissions };
+  }
+
+  public async getIdentityPermissionsByIdentity(
+    getIdentityPermissionsByIdentityRequest: GetIdentityPermissionsByIdentityRequest
+  ): Promise<GetIdentityPermissionsByIdentityResponse> {
+    this._isInit();
+    const identityPermissionsGroupKey = {
+      name: 'sk',
+      value: this._composeIdentityKey(
+        this._identityPermissionsPrefix,
+        getIdentityPermissionsByIdentityRequest.identityType,
+        getIdentityPermissionsByIdentityRequest.identityId
+      )
+    };
+    const identityPermissionsResponse = await this._awsService.helpers.ddb
+      .query({
+        index: this._getResourceBySortKeyIndex,
+        key: identityPermissionsGroupKey
+      })
+      .execute();
+
+    const identityPermissions: IdentityPermission[] =
+      identityPermissionsResponse.Items?.map((item) => {
+        return this._processItemToIdentityPermission(item);
+      }) ?? [];
+
+    return { identityPermissions };
+  }
+
+  private _isInit(): void {
+    if (!this._initializationPromise) {
+      throw new BadConfigurationError('Not Initialized');
+    }
+  }
+
+  private _composeIdentityKey(prefix: string, identityType: IdentityType, identityId: string): string {
+    return this._composeKey(prefix, identityType, identityId);
+  }
+
+  private _composeKey(prefix: string, type: string, id: string): string {
+    return `${prefix}#${type}#${id}`;
+  }
+
+  private _decomposeKey(key: string): { prefix: string; type: string; id: string } {
+    const split = key.split('#');
+    return {
+      prefix: split[0],
+      type: split[1],
+      id: split[2]
+    };
+  }
+
+  private _createItemFromIdentityPermission(identityPermission: IdentityPermission): Item {
+    const action = `${identityPermission.action}`;
+    const effect = `${identityPermission.effect}`;
+    const pk = this._composeKey(
+      this._identityPermissionsPrefix,
+      identityPermission.subjectType,
+      identityPermission.subjectId
+    );
+    const sk = this._composeIdentityKey(
+      this._identityPermissionsPrefix,
+      identityPermission.identityType,
+      identityPermission.identityId
+    );
+    // Same as sort key in order to keep identity type and identityid merged
+    const identity = sk;
+
+    const conditions = identityPermission.conditions ?? {};
+    const fields = identityPermission.fields ?? [];
+    return {
+      pk,
+      sk,
+      action,
+      effect,
+      subjectType: identityPermission.subjectType,
+      subjectId: identityPermission.subjectId,
+      identity,
+      conditions,
+      fields
+    };
+  }
+
+  private _processItemToIdentityPermission(item: Item): IdentityPermission {
+    const { type, id } = this._decomposeKey(item.identity as string);
```

Comment:
do we need `as string` now that item is typed?

Reply:
abstracted away using a DBItem